### PR TITLE
Remove ulele/deepcopier in favor of JSON deep copy

### PR DESCRIPTION
**Pull request by mheon**
We have a very high performance JSON library that doesn't need to perform code generation. Let's use it instead of our questionably performant, reflection-dependent deep copy library.

Most changes because some functions can now return errors.

Also converts cmd/podman to use jsoniter, instead of pkg/json, for increased performance.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
All kinds of test unhappiness @mheon

**Comment by mheon** [comment]
It's our good old friend `jsoniterator`, which has a firm belief that it should panic the program if someone tries to indent JSON with tabs, not spaces.

**Comment by mheon** [comment]
Fixed by replacing a `\t` with five spaces

File: cmd/podman/ps.go [hunk 1]
```diff
@@ -647,7 +646,7 @@ func printFormat(format string, containers []shared.PsContainerOutput) error {
 }
 
 func dumpJSON(containers []shared.PsContainerOutput) error {
-	b, err := json.MarshalIndent(containers, "", "\t")
+	b, err := json.MarshalIndent(containers, "", "     ")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/ps.go]: ugh

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, tests are getting a lot closer, but still not all in yet.

**Comment by mheon** [comment]
Rebased, CI should be happy now

**Pull request merged**